### PR TITLE
Fix Bugs of TE/JAX

### PR DESCRIPTION
**Pull request by mingxu1067**
1. Added support to `transpose_batch_seqlen` when `decoded=True`.
2. Fixed wrong masking issues in MultiHeadAttention when `decoded=True`.
3. Fixed missing dropout broadcast dims in LayerNormMLP.

### CODE REVIEW

**Comment by jeng1220** [comment]
@ptrendx and @timmoon10 ,
It is a hotfix for T5X model training.

File: transformer_engine/jax/transformer.py [hunk 1]
```diff
@@ -447,41 +447,38 @@ def kv_init(key, shape, dtype):
...
                 cached_key.value = key
                 cached_value.value = value
                 cache_index.value = cache_index.value + 1
 
-                key = jnp.moveaxis(key, -1, -3)
-                value = jnp.moveaxis(value, -1, -3)
-
                 mask = combine_masks(
-                    mask, jnp.broadcast_to(jnp.arange(length) <= cur_index, (batch, 1, 1, length)))
+                    mask, jnp.broadcast_to(jnp.arange(length) > cur_index, (batch, 1, 1, length)))
```
Comment (timmoon10) [inline transformer_engine/jax/transformer.py:481]: Can you explain your fix here? I'd imagine you'd want the attention mask to exclude tokens past `cur_index` since they haven't been initialized yet, but here you're only considering tokens past `cur_index`.

**Review by timmoon10 — APPROVED** [review]
Seems reasonable to me, but I'd like to better understand the bug you're fixing.

File: transformer_engine/jax/transformer.py (hunk 1 above)
Comment (mingxu1067) [reply transformer_engine/jax/transformer.py:481]: This is because value of masks has different meaning in custom softmax and original implementatino. In custom softmax, 1 means masking out and 0 means keep values, which is on the contrary with original implementation, therefore we have to revert the mask generation here to make inference works.

**Review by timmoon10 — APPROVED** [review]
